web-src/src/components/GenerateButton.js

```diff
@@ -50,6 +50,69 @@ export function GenerateButton() {
   const saveResults = useSaveResults();
   const { formatMessage } = useIntl();
 
+  const closeToast = () => {
+    const toast = document.querySelector('div[class*="spectrum-Toast-buttons"]');
+    if (toast !== null) {
+      const closeBtn = toast.querySelector('button[aria-label="Close"]');
+      if (closeBtn !== null) {
+        closeBtn.click();
+      }
+    }
+  };
+
+  const setLoadingToastMessages = () => {
+    const timeoutIds = [];
+    timeoutIds.push(
+      setTimeout(() => {
+        closeToast();
+        ToastQueue.info(
+          formatMessage(intlMessages.promptSessionSideView.variationsGeneration15SecondsDelayToast),
+          { timeout: 1500 },
+        );
+      }, 15000),
+    );
+    timeoutIds.push(
+      setTimeout(() => {
+        closeToast();
+        ToastQueue.info(
+          formatMessage(intlMessages.promptSessionSideView.variationsGeneration30SecondsDelayToast),
+          { timeout: 1500 },
+        );
+      }, 30000),
+    );
+    timeoutIds.push(
+      setTimeout(() => {
+        closeToast();
+        ToastQueue.info(
+          formatMessage(intlMessages.promptSessionSideView.variationsGeneration60SecondsDelayToast),
+          { timeout: 1500 },
+        );
+      }, 60000),
+    );
```

Comment:
This could be refactored into a loop if code readability is not impacted.

Reply:
Initially I thought to display uniq messages after 15, 30, 60 seconds using `setTimeout`, and after that display a general message using `setInterval`, now with suggestions from https://cq-dev.slack.com/archives/C050XSQ2BCN/p1715789435982609?thread_ts=1715785504.643539&cid=C050XSQ2BCN it is better to switch to the loop 💯 👍